### DISCUSSION TITLE
How to access resource routes in UI routes

### DISCUSSION THREAD

**Discussion by justin-hackin** [remix-run/remix#9032] [Q&A] [answered]
I am trying to access a resource route for a POST endpoint in my ui route loader. When I attempt to `fetch('/my-route')` I get an invalid URL error. I have to prepend `http://localhost:3000` for the fetch to work but I need an environment-agnostic way to construct the url.

The docs for Resource Routes feel incomplete because they do not show you how to consume the endpoints in any way. I was not able to find any articles about this matter online. 

Usually bundling libraries have base url constants or functions for constructing a url based upon the environment so I'm wondering if those exist for remix.

**Comment by justin-hackin** [comment]
I understand [this](https://github.com/remix-run/remix/discussions/5349) is related but instead of inquiring about supporting relative urls, I'm just looking for any solution to this more general issue.

**Reply by kiliman** [reply, marked as answer]
So the `fetch` is from your loader? Are you trying to call an endpoint in the current app?

You should always use an _absolute_ URL in `fetch` for _external_ server-side fetches.

However, if you're calling an existing `loader` in your own app, extracting that to a shared function you call would be better. In other words, don't make a _network_ request to an _internal_ route.

**Reply by justin-hackin** [reply]
For anyone who wants to see this point spelled out in detail, here's an example post: https://sergiodxa.com/tutorials/expose-remix-routes-as-api-endpoints
